src/Sarif.Multitool/Rules/SARIF2010.ProvideCodeSnippets.cs

```diff
@@ -65,5 +78,46 @@ private void AnalyzeRegion(Region region, string regionPointer)
                     nameof(RuleResources.SARIF2010_ProvideCodeSnippets_Note_Default_Text));
             }
         }
+
+        private bool AnalyzeArtifactLocation(ArtifactLocation artifactLocation)
+        {
+            // no artifactLocation / no artifacts, so we should look for the snippet
+            if (artifactLocation == null || this.artifacts == null)
+            {
+                return true;
+            }
+
+            artifactLocation.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri resolvedUri);
+
+            foreach (Artifact artifact in this.artifacts)
+            {
+                // content/text doesn't exist, continue to next
+                if (string.IsNullOrEmpty(artifact.Contents?.Text))
+                {
+                    continue;
+                }
+
+                artifact.Location.TryReconstructAbsoluteUri(this.originalUriBaseIds, out Uri artifactUri);
+
+                if (resolvedUri != null && artifactUri != null)
+                {
+                    if (artifactUri.AbsolutePath.Equals(resolvedUri.AbsolutePath, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    // couldn't generate the absoluteUri's, so let's compare everything
```

Comment:
>c [](start = 23, length = 1)

A small thing: Please begin each comment sentence with a capital letter, and end with a period. Thanks.